jcl/src/jdk.management/share/classes/openj9/lang/management/internal/OpenJ9DiagnosticsMXBeanImpl.java

```diff
@@ -0,0 +1,422 @@
+		checkManagementSecurityPermission();
+		switch (dumpAgent) {
+/*[IF Sidecar19-SE]*/
+		case "java": //$NON-NLS-1$
+			try {
+				fileName = (String)dump_javaDumpToFile.invoke(null, fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "heap": //$NON-NLS-1$
+			try {
+				fileName = (String)dump_heapDumpToFile.invoke(null, fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "system": //$NON-NLS-1$
+			try {
+				fileName = (String)dump_systemDumpToFile.invoke(null, fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "snap": //$NON-NLS-1$
+			try {
+				fileName = (String)dump_snapDumpToFile.invoke(null, fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+/*[ELSE]
+		case "java": //$NON-NLS-1$
+			try {
+				fileName = Dump.javaDumpToFile(fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "heap": //$NON-NLS-1$
+			try {
+				fileName = Dump.heapDumpToFile(fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "system": //$NON-NLS-1$
+			try {
+				fileName = Dump.systemDumpToFile(fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "snap": //$NON-NLS-1$
+			try {
+				fileName = Dump.snapDumpToFile(fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+/*[ENDIF]*/
+		default:
+			/*[MSG "K0663", "Invalid or Unsupported Dump Agent cannot be triggered"]*/
+			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0663")); //$NON-NLS-1$
+		}
+		return fileName;
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String triggerClassicHeapDump() throws InvalidOptionException {
+		String dumpOptions = "heap:opts=CLASSIC"; //$NON-NLS-1$
+		checkManagementSecurityPermission();
+		try {
+/*[IF Sidecar19-SE]*/
+			String fileName = (String)dump_triggerDump.invoke(null, dumpOptions);
+/*[ELSE]
+			String fileName = Dump.triggerDump(dumpOptions);
+/*[ENDIF]*/
+			return fileName;
+		} catch (Exception e) {
+
+			handleInvalidDumpOptionException(e);
+			throw handleError(e);
+		}
+	}
+
+	private static void checkManagementSecurityPermission() {
+		/* Check the caller has Management Permission. */
+		SecurityManager manager = System.getSecurityManager();
+		if( manager != null ) {
+/*[IF Sidecar19-SE]*/
+			try {
+				Object controlPermission = managementPermission.get(null);
+				manager.checkPermission((java.security.Permission)controlPermission);
```

Comment:
Please pull the call to checkPermission() out of the `try` statement so it doesn't flow through `handleError` and is common for all Java versions.